File: Rust/src/node.rs
pub struct Node {
    pub x: i32,
    pub y: i32,
    pub next: Option<Box<Node>>,
}

impl Node {
    pub fn new(x: i32, y: i32, next: Option<Box<Node>>) -> Self {
        Node { x, y, next }
    }
}

File: Typescript/src/node.ts
export default class Node implements Iterable<Node> {
    constructor(
        public x: number,
        public y: number,
        public next: Node | null,
    ) {}

    *[Symbol.iterator](): Iterator<Node> {
        let curr: Node | null = this;
        while (curr) {
            yield curr;
            curr = curr.next;
        }
    }
}
